test/fakes/fakeupstreamauthority/fakeupstreamauthority.go

```diff
@@ -134,11 +139,29 @@ func certsDER(certs []*x509.Certificate) [][]byte {
 	return out
 }
 
-func (m *UpstreamAuthority) PublishJWTKey(context.Context, *upstreamauthority.PublishJWTKeyRequest) (*upstreamauthority.PublishJWTKeyResponse, error) {
+func (m *UpstreamAuthority) PublishJWTKey(req *upstreamauthority.PublishJWTKeyRequest, stream upstreamauthority.UpstreamAuthority_PublishJWTKeyServer) error {
 	switch {
 	case m.config.PublishJWTKeyResponse != nil:
-		return m.config.PublishJWTKeyResponse, nil
+		return stream.Send(m.config.PublishJWTKeyResponse)
 	default:
-		return nil, status.Errorf(codes.Unimplemented, "unimplemented on fake")
+		return status.Errorf(codes.Unimplemented, "unimplemented on fake")
 	}
 }
+
+func (m *UpstreamAuthority) Configure(context.Context, *plugin.ConfigureRequest) (*plugin.ConfigureResponse, error) {
+	return &plugin.ConfigureResponse{}, nil
+}
+
+func (m *UpstreamAuthority) GetPluginInfo(context.Context, *plugin.GetPluginInfoRequest) (*plugin.GetPluginInfoResponse, error) {
+	return &plugin.GetPluginInfoResponse{}, nil
+}
+
+func Load(t *testing.T, config Config) (upstreamauthority.UpstreamAuthority, *UpstreamAuthority, func()) {
```

Comment:
This would be a good opportunity to leverage `t.Cleanup` so the caller doesn't have to invoke the "done" function to clean up the plugin.